Adds new exosuit modules

CrimsonShrike opened the pull request:
🆑CrimsonShrike
rscadd: Adds exosuit flash.
rscadd: Adds exosuit camera.
tweak: Exosuit floodlights can be mounted on shoulders now too.
/🆑

As usual. sprites courtesy of Azlan.

CrimsonShrike commented on code/modules/mechs/equipment/combat.dm:
```diff
@@ -443,3 +444,99 @@
 			playsound(user.loc, 'sound/weapons/Genhit.ogg', 50, 1)
 			return AURA_FALSE|AURA_CANCEL
+
+/obj/item/mech_equipment/flash
```
Flash code is hell. Tried to make them a bit better than handheld ones, but nowhere near as op as floor ones. If you want to change values suggest away.

Spookerton replied:
Not sure about the forced drop, but I guess we'll see how it plays out.

Tennessee116 commented on code/modules/mechs/equipment/utility.dm:
```diff
@@ -685,3 +689,82 @@
+	equipment_delay = 10
+
+	origin_tech = list(TECH_MATERIAL = 1, TECH_ENGINEERING = 2, TECH_MAGNET = 2)
+	var/obj/machinery/camera/network/thunder/camera
```
Wouldn't this work better in the `NETWORK_ROBOTS`? I do understand that there isn't a preset for cameras on that network, though.

CrimsonShrike replied:
Is on a public network by default so that people dont become unable to use it. People can then set it to whatever they feel like.

SierraKomodo commented on code/modules/mechs/equipment/utility.dm:
```diff
@@ -685,3 +689,82 @@
+/obj/item/mech_equipment/camera/examine(mob/user)
+	. = ..()
+	to_chat(user, "Network: [english_list(camera.network)]; Feed is currently: [camera.status ? "Online" : "Offline"].")
+	if(user.skill_check(SKILL_DEVICES, SKILL_BASIC))
+		to_chat(user, SPAN_SUBTLE("You could modify the output network using a screwdriver."))
```
Unless the interaction itself is also locked behind a skill check, there's not much point in hiding this information.

This would also probably be better suited in a codex entry.

SierraKomodo commented on code/modules/mechs/equipment/utility.dm:
```diff
@@ -685,3 +689,82 @@
 		else
 			to_chat(user, SPAN_WARNING("You cannot slide there!"))
+
+/obj/item/mech_equipment/camera
```
Not going to make it a hard requirement right now but I'd prefer new items get codex entries. I'd suggest a new file for `mech_equipment.dm` or something to that effect in `code\modules\codex\entries` for these.

SierraKomodo commented on code/modules/mechs/equipment/combat.dm:
```diff
@@ -443,3 +444,99 @@
 			playsound(user.loc, 'sound/weapons/Genhit.ogg', 50, 1)
 			return AURA_FALSE|AURA_CANCEL
+
+/obj/item/mech_equipment/flash
```
Same thing re: codex

SierraKomodo reviewed (changes requested):
Haven't done a proper test run yet.

SierraKomodo commented on code/modules/codex/entries/exosuits.dm:
```diff
@@ -0,0 +1,16 @@
+
+/datum/codex_entry/exosuit_equipment
+	associated_paths = list(/obj/item/mech_equipment)
+	mechanics_text = "Exosuit equipment is mounted onto specific exosuit hardpoints. In order to correctly attach itself, it will also need the exosuit to have the right software \
```
```suggestion
	mechanics_text = "Exosuit equipment is mounted onto specific exosuit hardpoints. In order to correctly attach itself, it will also need the exosuit to have the right software. \
```

Spookerton commented on code/modules/mechs/equipment/utility.dm:
```diff
@@ -685,3 +689,80 @@
+	. = ..()
+	
+	if(isScrewdriver(W))
+		var/list/all_networks[0]
```
```suggestion
		var/list/all_networks = list()
```
as much as I wish we'd settled on [], style is list() except for arcane applications.

CrimsonShrike committed:
Adds new exosuit modules


Apply suggestions from code review

Co-authored-by: Sierra Brown (SierraKomodo) <SierraKomodo@users.noreply.github.com>
Update code/modules/mechs/equipment/utility.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
